catalogd/internal/serverutil/serverutil_tests.go

```diff
@@ -0,0 +1,254 @@
+package serverutil
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	"sigs.k8s.io/yaml"
+)
+
+// {
+// 	name: "provides gzipped content for the path /catalogs/test-catalog/api/v1/all with size > 1400 bytes",
+// 	setupStore: func() error {
+// 		return writeFile(filepath.Join(store.RootDir, "test-catalog3", "catalog.jsonl"), []byte(testCompressableJSON), 0600)
+// 	},
+// 	expectStatusOK:  true,
+// 	expectedContent: testCompressableJSON,
+// 	URLPath:         "/catalogs/test-catalog3/api/v1/all",
+// },
+
+// jsonLineFormattedCompresableJSON, err := generateJSONLines([]byte(testCompressableJSON))
+// 	require.NoError(t, err)
+// {
+// 	name: "Provides JSON-lines format for the served JSON catalog",
+// 	setupStore: func() error {
+// 		unpackResultFS := &fstest.MapFS{
+// 			"catalog.json": &fstest.MapFile{Data: []byte(testCompressableJSON), Mode: os.ModePerm},
+// 		}
+// 		return store.Store(context.Background(), "test-catalog4", unpackResultFS)
+// 	},
+// 	expectStatusOK:  true,
+// 	expectedContent: jsonLineFormattedCompresableJSON,
+// 	URLPath:         "/catalogs/test-catalog4/api/v1/all",
+// },
+
+// yamlData, err := makeYAMLFromConcatenatedJSON([]byte(testCompressableJSON))
+// require.NoError(t, err)
+// jsonLineFormattedYamlData, err := generateJSONLines(yamlData)
+// require.NoError(t, err)
+// {
+// 	name:            "Provides JSON-lines format for the served YAML catalog",
+// 	expectStatusOK:  true,
+// 	expectedContent: jsonLineFormattedYamlData,
+// 	URLPath:         fmt.Sprintf("%s/test-catalog/api/v1/all", urlPrefix),
+// 	setupStore: func() error {
+// 		yamlData, err := makeYAMLFromConcatenatedJSON([]byte(testCompressableJSON))
+// 		if err != nil {
+// 			return err
+// 		}
+// 		unpackResultFS := &fstest.MapFS{
+// 			"catalog.yaml": &fstest.MapFile{Data: yamlData, Mode: os.ModePerm},
+// 		}
+// 		err = store.Store(context.Background(), "test-catalog", unpackResultFS)
+// 		if err != nil {
+// 			return err
+// 		}
+// 		return err
+// 	},
+// },
+
+// {
+// 	name: "Ignores accept-encoding for the path /catalogs/test-catalog/api/v1/all with size < 1400 bytes",
+// 	setupStore: func() error {
+// 		return writeFile(filepath.Join(store.RootDir, "test-catalog2", "catalog.jsonl"), []byte(`{"foo":"bar"}`), 0600)
+// 	},
+// 	expectStatusOK:  true,
+// 	expectedContent: `{"foo":"bar"}`,
+// 	URLPath:         "/catalogs/test-catalog2/api/v1/all",
+// },
+
+// by default the compressor will only trigger for files larger than 1400 bytes
+const testCompressableJSON = `{
```

Comment:
If all this is stuff is to test the gzip handler in isolation, I think we can simplify.

If the goal is to test the entire stack of handlers, I figure there's a better way to organize and test to reduce duplication and fragility. For example, if we make the `CatalogServerConfig` accept an arbitrary `http.Handler` instead of a `storage.Instance`, then we can:
- test the `storage.Instance` handler in isolation
- test the `serverutil` setup with dummy handlers that are easier to manipulate for test purposes.

Reply:
Or, the other relevant question is, do we need to test the gzip handler explicitly in our unit tests? We're importing `"github.com/klauspost/compress/gzhttp"` to provided compressed content via our server. Are unit tests that say "tests if the library is doing its job" really that useful?

Reply:
IMHO, any external interaction is a good test.  And unit tests are better than e2e (smaller, faster, usually more tightly-scoped).
We /were/ testing to ensure that we would respond accordingly for `accept-encoding` request headers that could control/enable this functionality.

Reply:
Added these tests in `serverutil_tests.go`